pkg/middlewares/ratelimiter/redisrate/lua.go

```diff
@@ -0,0 +1,56 @@
+package redisrate
+
+import "github.com/redis/go-redis/v9"
+
+//nolint:dupword
+var allowTokenBucket = redis.NewScript(`
+local key = KEYS[1]
+local limit, burst, ttl, t, n, max_delay = tonumber(ARGV[1]), tonumber(ARGV[2]), tonumber(ARGV[3]), tonumber(ARGV[4]),
+    tonumber(ARGV[5]), tonumber(ARGV[6])
+
+if n > burst then
+    return {tostring(false), tostring(0)}
+end
+
+local bucket = {
+    limit = limit,
+    burst = burst,
+    tokens = 0,
+    last = 0
+}
+
+local rl_source = redis.call('hgetall', key)
+
+-- TODO: check case limit inf and equal to 0.
+if table.maxn(rl_source) == 4 then
+    -- Get bucket state from redis
+    bucket.last = tonumber(rl_source[2])
+    bucket.tokens = tonumber(rl_source[4])
+end
+
+-- TODO: edge cases for later.
```

Comment:
What are the edge cases you have in mind?

Reply:
Sorry this is just a dummy comment I forgot to remove.